src/components/FiscalYear/EditFiscalYear/EditFiscalYear.js

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import ReactRouterPropTypes from 'react-router-prop-types';
+import { withRouter } from 'react-router-dom';
+import { get } from 'lodash';
+
+import { stripesConnect } from '@folio/stripes/core';
+import { Paneset } from '@folio/stripes/components';
+import {
+  LoadingPane,
+  useShowToast,
+} from '@folio/stripes-acq-components';
+
+import {
+  FISCAL_YEAR_VIEW_ROUTE,
+} from '../../../common/const';
+import {
+  fiscalYearResource,
+} from '../../../common/resources';
+import FiscalYearForm from '../FiscalYearForm';
+
+const EditFiscalYear = ({ resources, mutator, match, history }) => {
+  const fiscalYearId = match.params.id;
+
+  useEffect(
+    () => {
+      mutator.fiscalYearEdit.reset();
+      mutator.fiscalYearEdit.GET();
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [fiscalYearId],
+  );
+
+  const showToast = useShowToast();
+
+  const closeEdit = useCallback(
+    () => {
+      history.push(`${FISCAL_YEAR_VIEW_ROUTE}${fiscalYearId}?layer=view`);
+    },
+    [fiscalYearId, history],
+  );
+
+  const saveFiscalYear = useCallback(
+    async (fiscalYearValues) => {
+      try {
+        const savedFiscalYear = await mutator.fiscalYearEdit.PUT(fiscalYearValues);
+
+        showToast('ui-finance.fiscalYear.actions.save.success');
+        setTimeout(() => closeEdit(), 0);
```

Comment:
what the reason of timeout here?

Reply:
why it can't be something simple, as in invoice? are there any issues? https://github.com/folio-org/ui-invoice/pull/110/files#diff-92285e71ea65dcce6feaa47847737e9cR42

Reply:
To prevent pop-up that you want to leave the form page with unsaved data. In invoices app we use an approach with layers instead of diff routes